sky/skylet/providers/kubernetes/node_provider.py

```diff
@@ -209,6 +210,31 @@ def create_node(self, node_config, tags, count):
         start = time.time()
         while True:
             if time.time() - start > self.timeout:
+                pod_status = new_nodes[0].status.phase
+                pod_name = new_nodes[0]._metadata._name
+                events = kubernetes.core_api().list_namespaced_event(
+                    self.namespace,
+                    field_selector=f"involvedObject.name={pod_name}")
+                for event in events.items:
+                    if event.reason == 'FailedScheduling':
+                        event_message = event.message
+                        break
+                if pod_status == 'Pending':
+                    if event_message is not None:
+                        if 'Insufficient cpu' in event_message:
+                            total_cpus = os.cpu_count()
+                            raise config.KubernetesError(
+                                'More than available CPU(s) are requested. '
+                                f'In total, {total_cpus} CPUs are available in '
+                                'the machine. Run \'sky status\' to see the '
+                                'number of CPUs that are already in use.')
+                        if 'didn\'t match Pod\'s node affinity/selector' in event_message:
+                            node_selector = pod.spec.node_selector
+                            if node_selector is not None:
+                                raise config.KubernetesError(
+                                    'Unavailable GPU(s) are requested. '
+                                    f'Please confirm if {node_selector} is '
+                                    'available in the cluster.')
```

Comment:
Can we still print the event_message if it is neither of these two reasons? For example, there can be many other reasons - disk pressure, memory pressure, all nodes are tainted... etc.  It'll be useful for debugging and helping understand why scheduling failed

Reply:
@romilbhardwaj fixed!

Reply:
I think other generic errors are still not surfaced. For example, I tainted a node to not allow any scheduling on it:
```
kubectl taint nodes <node-name> key=value:NoSchedule
``` 
Then ran `sky launch`, which gave `More than available Memories are requested.`. My cluster has enough memory and CPU:
```
$ sky launch -c test

...
W 08-27 08:52:32 cloud_vm_ray_backend.py:975] Got error(s) in kubernetes:
W 08-27 08:52:32 cloud_vm_ray_backend.py:977] 	sky.skylet.providers.kubernetes.config.KubernetesError: More than available Memories are requested. Run 'sky status' to see the amount of Memories that are already in use.
W 08-27 08:52:45 cloud_vm_ray_backend.py:2123] sky.exceptions.ResourcesUnavailableError: Failed to acquire resources in all zones in kubernetes. Try changing resource requirements or use another region.
...
```

For reference, the actual error is correctly displayed in`kubectl describe pods`:
```
Events:
  Type     Reason            Age   From               Message
  ----     ------            ----  ----               -------
  Warning  FailedScheduling  2s    default-scheduler  0/1 nodes are available: 1 node(s) had untolerated taint {key: value}. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling..
```
Full logs [here](https://gist.github.com/romilbhardwaj/68750b69f8ceb9f31a886dae8d8f8a72).

Reply:
Memory related error message will be printed out only when `'Insufficient memory'` is in the error message from `kubectl describe pods`.

Fixed and confirmed the case above now surfaces the correct error message.
```
W 08-29 01:56:11 cloud_vm_ray_backend.py:975]   sky.skylet.providers.kubernetes.config.KubernetesError: Timed out while waiting for nodes to start. Cluster may be out of resources or may be too slow to autoscale. Error: 0/2 nodes are available: 1 node(s) didn't match Pod's node affinity/selector, 1 node(s) had untolerated taint {key: value}. preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling..
```

Currently, other generic errors that are not due to lack of CPUs, Memory, GPUs will be handled by the following:
```
                    raise config.KubernetesError(
                        f'{timeout_err_msg} '
                        f'Error: {event_message}')
                raise config.KubernetesError(f'{timeout_err_msg}')
```